### PR TITLE
RIA-2794: End to end flow

### PR DESCRIPTION
**Pull request by divisathyan**
Add JUnit test case

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/hois/controllers/HoisController.java [hunk 1]
```diff
@@ -30,16 +37,22 @@
 @RestController
 public class HoisController {
 
+    @Autowired
+    private transient ObjectMapper objectMapper;
+
+    @org.springframework.beans.factory.annotation.Value("classpath:homeoffice-api-response-mock.json")
```
Comment (iguannago) [inline src/main/java/uk/gov/hmcts/reform/hois/controllers/HoisController.java]: this is production code, I mean, it's a test class but you are naming this resourceResFile with the mock word in it. It seems a bit weird to a mock on code that goes to production. Is this OK?

File: src/main/java/uk/gov/hmcts/reform/hois/domain/entities/DecisionCommunication.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
+package uk.gov.hmcts.reform.hois.domain.entities;
+
+import com.fasterxml.jackson.databind.PropertyNamingStrategy;
+import com.fasterxml.jackson.databind.annotation.JsonNaming;
+import lombok.Data;
+
+@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
+@Data
```
Comment (iguannago) [inline src/main/java/uk/gov/hmcts/reform/hois/domain/entities/DecisionCommunication.java]: if you use the Data annotation then you can skip all the getters and setters boilerplate code.

File: src/main/java/uk/gov/hmcts/reform/hois/domain/entities/ErrorDetail.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
+package uk.gov.hmcts.reform.hois.domain.entities;
+
+import com.fasterxml.jackson.databind.PropertyNamingStrategy;
+import com.fasterxml.jackson.databind.annotation.JsonNaming;
+import lombok.Data;
+
+@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
+@Data
```
Comment (iguannago) [inline src/main/java/uk/gov/hmcts/reform/hois/domain/entities/ErrorDetail.java]: same here as above

File: src/main/java/uk/gov/hmcts/reform/hois/domain/entities/FieldType.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
+package uk.gov.hmcts.reform.hois.domain.entities;
+
+import com.fasterxml.jackson.databind.PropertyNamingStrategy;
+import com.fasterxml.jackson.databind.annotation.JsonNaming;
+import lombok.Data;
+
+@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
+@Data
```
Comment (iguannago) [inline src/main/java/uk/gov/hmcts/reform/hois/domain/entities/FieldType.java]: same

File: src/main/java/uk/gov/hmcts/reform/hois/domain/entities/HomeOfficeRequest.java [hunk 5]
```diff
@@ -3,17 +3,24 @@
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.fasterxml.jackson.databind.PropertyNamingStrategy;
 import com.fasterxml.jackson.databind.annotation.JsonNaming;
+import lombok.Data;
 
 @JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
+@Data
```
Comment (iguannago) [inline src/main/java/uk/gov/hmcts/reform/hois/domain/entities/HomeOfficeRequest.java]: the same

File: src/main/java/uk/gov/hmcts/reform/hois/domain/entities/HomeOfficeResponse.java [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.PropertyNamingStrategy;
+import com.fasterxml.jackson.databind.annotation.JsonNaming;
+import java.util.Arrays;
+import lombok.Data;
+
+@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
+@Data
+public class HomeOfficeResponse {
+    /* @JsonProperty("ho_reference")
```
Comment (iguannago) [inline src/main/java/uk/gov/hmcts/reform/hois/domain/entities/HomeOfficeResponse.java]: you can safely clean up these commented lines. You can see the history and bring it up at any point if needed.

File: src/main/java/uk/gov/hmcts/reform/hois/domain/entities/HomeOfficeResponse.java [hunk 7]
```diff
@@ -0,0 +1,77 @@
+package uk.gov.hmcts.reform.hois.domain.entities;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.PropertyNamingStrategy;
+import com.fasterxml.jackson.databind.annotation.JsonNaming;
+import java.util.Arrays;
+import lombok.Data;
+
+@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
+@Data
```
Comment (iguannago) [inline src/main/java/uk/gov/hmcts/reform/hois/domain/entities/HomeOfficeResponse.java]: the same thing

File: src/main/java/uk/gov/hmcts/reform/hois/domain/entities/IAhoisResponse.java [hunk 8]
```diff
@@ -0,0 +1,46 @@
+package uk.gov.hmcts.reform.hois.domain.entities;
+
+import com.fasterxml.jackson.databind.PropertyNamingStrategy;
+import com.fasterxml.jackson.databind.annotation.JsonNaming;
+import lombok.Data;
+
+@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
+@Data
```
Comment (iguannago) [inline src/main/java/uk/gov/hmcts/reform/hois/domain/entities/IAhoisResponse.java]: the same thing

File: build.gradle [hunk 9]
```diff
@@ -276,6 +276,8 @@ dependencies {
   testRuntimeOnly group: 'org.junit.jupiter', name: 'junit-jupiter-engine', version: versions.junit
 
   testCompile group: 'org.springframework.boot', name: 'spring-boot-starter-test'
+  testCompile "com.github.tomakehurst:wiremock:2.19.0"
```
Comment (iguannago) [inline build.gradle]: Not sure why wiremock is needed you can achieve the same by just using SpringbootTEst

File: src/main/java/uk/gov/hmcts/reform/hois/controllers/HoisController.java (hunk 1 above)
Comment (divisathyan) [reply src/main/java/uk/gov/hmcts/reform/hois/controllers/HoisController.java]: As of now, we do not have the external API from HomeOffice. For now, we have decided to stub it.

File: src/main/java/uk/gov/hmcts/reform/hois/domain/entities/DecisionCommunication.java (hunk 2 above)
Comment (divisathyan) [reply src/main/java/uk/gov/hmcts/reform/hois/domain/entities/DecisionCommunication.java]: Tried it but lambok and jackson does not work together. Working on it

File: src/main/java/uk/gov/hmcts/reform/hois/domain/entities/HomeOfficeResponse.java (hunk 6 above)
Comment (divisathyan) [reply src/main/java/uk/gov/hmcts/reform/hois/domain/entities/HomeOfficeResponse.java]: Removed as suggested

File: build.gradle (hunk 9 above)
Comment (divisathyan) [reply build.gradle]: I used the ia-case-api as the base. I am imitating an external API built by HomeOffice and it is essential to mock the response.

**Pull request closed without merging**